### PR TITLE
Improve API doc for geometry/distance.ts

### PR DESCRIPTION
**Pull request by dumganhar**
Re: #13888

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/core/geometry/distance.ts [hunk 1]
```diff
@@ -39,26 +39,26 @@ const e = new Array(3);
...
- * the distance between a point and a plane
+ * Calculates the distance between a point and a plane.
  * @zh
  * 计算点和平面之间的距离。
- * @param {Vec3} point @en Target point @zh 目标点。
- * @param {Plane} plane @en Target plane @zh 目标平面。
- * @return @en The distance in between @zh 距离。
+ * @param point @en The target point. @zh 目标点。
+ * @param plane @en The target plane. @zh 目标平面。
+ * @returns @en The distance between the point and the plane. @zh 点和平面之间的距离。
```
Comment (minggo) [inline cocos/core/geometry/distance.ts:47]: I just wonder using `a` is better. For example
`The distance between a point and a plane.`

File: cocos/core/geometry/distance.ts [hunk 2]
```diff
@@ -39,26 +39,26 @@ const e = new Array(3);
...
  * @en
- * the closest point on plane to a given point
+ * Calculates the closest point on a plane to a given point.
  * @zh
  * 计算平面上最接近给定点的点。
- * @param out @en The closest point, as a result @zh 最近点。
- * @param point @en The given point @zh 给定点。
- * @param plane @en Target plane @zh 平面。
- * @return @en The closest point, same as out @zh 最近点。
+ * @param out @en The result of the closest point to store into. @zh 存储最近点的向量。
```
Comment (minggo) [inline cocos/core/geometry/distance.ts]: 是 `向量` 吗？

Comment (minggo) [inline cocos/core/geometry/distance.ts]: The result of the closest point to store into。-> The closest point.

File: cocos/core/geometry/distance.ts [hunk 3]
```diff
@@ -67,13 +67,13 @@ export function pt_point_plane (out: Vec3, point: Vec3, plane_: Plane) {
...
- * the closest point on aabb to a given point
+ * Calculates the closest point on an AABB to a given point.
  * @zh
- * 计算 aabb 上最接近给定点的点。
- * @param {Vec3} out @en The closest point, as a result @zh 最近点。
- * @param {Vec3} point @en The given point @zh 给定点。
- * @param {AABB} aabb @en Target aabb to calculate @zh 轴对齐包围盒。
- * @return {Vec3} @en The closest point, same as out @zh 最近点。
+ * 计算 AABB 上最接近给定点的点。
+ * @param out @en The result of the closest point to store into. @zh 存储最近点的向量。
```
Comment (minggo) [inline cocos/core/geometry/distance.ts]: The same as above.

File: cocos/core/geometry/distance.ts [hunk 4]
```diff
@@ -67,13 +67,13 @@ export function pt_point_plane (out: Vec3, point: Vec3, plane_: Plane) {
...
+ * @param point @en The given point @zh 给定点。
+ * @param aabb @en The target AABB to calculate. @zh 轴对齐包围盒。
```
Comment (minggo) [inline cocos/core/geometry/distance.ts]: The target AABB to calculate -> Axis alignment AABB to calculate.

File: cocos/core/geometry/distance.ts [hunk 5]
```diff
@@ -92,13 +92,13 @@ export function pt_point_aabb (out: Vec3, point: Vec3, aabb_: AABB): Vec3 {
 
 /**
  * @en
- * the closest point on obb to a given point
+ * Calculates the closest point on a OBB to a given point.
```
Comment (minggo) [inline cocos/core/geometry/distance.ts]: a OBB -> an OBB

File: cocos/core/geometry/distance.ts [hunk 6]
```diff
@@ -92,13 +92,13 @@ export function pt_point_aabb (out: Vec3, point: Vec3, aabb_: AABB): Vec3 {
...
  * @zh
- * 计算 obb 上最接近给定点的点。
- * @param {Vec3} out @en The closest point, as a result @zh 最近点。
- * @param {Vec3} point @en The given point @zh 给定点。
- * @param {OBB} obb @en Target obb to calculate @zh 方向包围盒。
- * @return {Vec3} @en The closest point, same as out @zh 最近点。
+ * 计算方向包围盒上最接近给定点的点。
+ * @param out @en The result of the closest point to store into. @zh 存储最近点的向量。
```
Comment (minggo) [inline cocos/core/geometry/distance.ts]: The same as above.

Comment (minggo) [reply cocos/core/geometry/distance.ts:98]: 方向包围盒 翻译有点别扭，干脆就别翻译，保留 OBB 就好了。

File: cocos/core/geometry/distance.ts [hunk 7]
```diff
@@ -92,13 +92,13 @@ export function pt_point_aabb (out: Vec3, point: Vec3, aabb_: AABB): Vec3 {
...
  * @en
- * the closest point on obb to a given point
+ * Calculates the closest point on a OBB to a given point.
  * @zh
- * 计算 obb 上最接近给定点的点。
- * @param {Vec3} out @en The closest point, as a result @zh 最近点。
- * @param {Vec3} point @en The given point @zh 给定点。
- * @param {OBB} obb @en Target obb to calculate @zh 方向包围盒。
- * @return {Vec3} @en The closest point, same as out @zh 最近点。
+ * 计算方向包围盒上最接近给定点的点。
```
Comment (minggo) [inline cocos/core/geometry/distance.ts]: 方向包围盒 翻译有点别扭，干脆就别翻译，保留 OBB 就好了。

File: cocos/core/geometry/distance.ts [hunk 8]
```diff
@@ -141,20 +141,20 @@ export function pt_point_obb (out: Vec3, point: Vec3, obb_: OBB): Vec3 {
...
- * Calculate the nearest point on the line, which is from A to B, to the given point
+ * Calculates the closest point on the line, which is from A to B, to the given point.
  * @zh
- * 计算给定点距离线段上最近的一点。线段从A到B。
- * @param out @en The closest point, as a result @zh 最近点
- * @param point @en The given point @zh 给定点
- * @param linePointA @en Point A on the line @zh 线上的某点 A
- * @param linePointB @en Point B on the line @zh 线上的某点 B
+ * 计算给定点距离线段 AB 上最近的一点。
+ * @param out @en The result of the closest point to store into. @zh 存储最近点的向量。
```
Comment (minggo) [inline cocos/core/geometry/distance.ts]: The same as above.

File: cocos/core/geometry/distance.ts (hunk 2 above)
Comment (dumganhar) [reply cocos/core/geometry/distance.ts]: Vec3 类型，Vector 不是翻译为向量？

**Review by minggo — APPROVED** [review]

**Pull request merged**